src/components/widgets/SelectWidget.js

```diff
@@ -7,7 +7,9 @@ import {asNumber} from "../../utils";
  * always retrieved as strings.
  */
 function processValue({type, items}, value) {
-  if (type === "array" && items && ["number", "integer"].includes(items.type)) {
+  if (value === "") {
+    return undefined;
```

Comment:
I added a note, but I'm not sure how well I explained it or if there's a better spot.